### ISSUE TITLE
punctuation and offlist rankings interactions

### ISSUE THREAD

**Issue by OllyBallance** [danballance/awesome-parser#26]
The attached files show how punctuation and rankings are currently interacting problematically:

https://www.dropbox.com/s/fy2h6j5daerljck/OB_BUGR_26.rar?dl=0

This can probably be handled in the short term by manually editing files and removing problematic punctuation, but seemed a good idea to have some relevant data recorded here in case it can be addressed at some future date.

**Comment by OllyBallance** [comment]
OK, it seems the only problems I now encounter I regex char escape problems like before. If I distill text files until I identify the problem character, I can delete, or presumably add it to the config as you showed me before. So things are good to, hazzah!

**Comment by OllyBallance** [comment]
So I've added to the config this character: |

Can run the corresponding prob text now - fingers crossed this'll work for all future probs

**Comment by danballance** [comment]
Okay cool. There's a good cheat sheet here with most of the characters
listed that have special meanings, it might help you:

http://www.cheatography.com/davechild/cheat-sheets/regular-expressions

**Comment by OllyBallance** [comment]
how you doing? fancy a chat?

On 7 November 2015 at 18:23, Dan Ballance <notifications@github.com> wrote:

> Okay cool. There's a good cheat sheet here with most of the characters
> listed that have special meanings, it might help you:
>
> http://www.cheatography.com/davechild/cheat-sheets/regular-expressions
>
> On Fri, 6 Nov 2015 20:46 OllyBallance <notifications@github.com> wrote:
>
> > So I've added to the config this character: |
> >
> > Can run the corresponding prob text now - fingers crossed this'll work
> for
> > all future probs
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/danballance/awesome-parser/issues/26#issuecomment-154528009
> >
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danballance/awesome-parser/issues/26#issuecomment-154624288>
> .
>



--

**Comment by OllyBallance** [comment]
ah cool - the two i've had problems with are both in Common Metach­ara­cters

On 7 November 2015 at 18:23, Dan Ballance <notifications@github.com> wrote:

> Okay cool. There's a good cheat sheet here with most of the characters
> listed that have special meanings, it might help you:
>
> http://www.cheatography.com/davechild/cheat-sheets/regular-expressions
>
> On Fri, 6 Nov 2015 20:46 OllyBallance <notifications@github.com> wrote:
>
> > So I've added to the config this character: |
> >
> > Can run the corresponding prob text now - fingers crossed this'll work
> for
> > all future probs
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/danballance/awesome-parser/issues/26#issuecomment-154528009
> >
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danballance/awesome-parser/issues/26#issuecomment-154624288>
> .
>



--

**Comment by danballance** [comment]
Are you able to tell me what your search word was for this run of the code? I need to try and replicate running it exactly on my machine.

**Comment by danballance** [comment]
Okay, single apostrophes - I've understood the issue here. I am currently allowing ' as part of words so that tokens for the n't part of don't are not excluded. I've come up with a much more complicated regex now, mostly taken from this stack overflow answer:

http://stackoverflow.com/a/18767749

The regex line I currently have in the config file is:

"regex_is_word": "^(?=([^']*[\\w]')?[^\\W]+$).{1,}$",

(notice this includes the trailing comma and which is part of the config format)

So basically this means that any double sequences of single quotes '' are treated as punctuation and not a word. These double  quote sequences actually have a special meaning in the Treebank parser and that was the underlying cause of this issue.

This improved regex also seems to have solved the trailing comma issue :)

**Comment by danballance** [comment]
(PS I'll update the config.json.example file with the definitive configuration when I'm done)

**Comment by OllyBallance** [comment]
hey, yep, should be 'as'

On 8 November 2015 at 06:39, Dan Ballance <notifications@github.com> wrote:

> Are you able to tell me what your search word was for this run of the
> code? I need to try and replicate running it exactly on my machine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danballance/awesome-parser/issues/26#issuecomment-154728912>
> .
>



--
